### PR TITLE
github workflows for publish and release

### PR DESCRIPTION
**Pull request by clenfest**

### CODE REVIEW

**Comment by clenfest** [comment]
Note to reviewers. Don't bother with the diff. Both of these files are new. I was dumb about how to test originally, but I figured it out.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,28 +1,44 @@
-name: Publish packages to npm
+name: Publish
 
 on:
```
Comment (dariuszkuc) [inline .github/workflows/publish.yml:3]: I think this should be just (no push)

```
on:
  release:
    types: [published]
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -22,8 +23,17 @@ jobs:
 
       - name: Install Dependencies
         run: npm i
+        
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
```
Comment (dariuszkuc) [inline .github/workflows/release.yml:29]: I believe `set-output` is deprecated -> https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

You should just use `GITHUB_OUTPUT`

```
- name: Set output
  run: echo "{name}={value}" >> $GITHUB_OUTPUT
```

Comment (clenfest) [reply .github/workflows/release.yml:29]: I checked the docs, it looks like it's available in `${{ github.ref_name }}` now without me needing to parse it.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
+    name: Publish slack webhook notifying about approval
+    if: "contains(github.event.head_commit.message, 'release:')"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Send to Slack
+        env: 
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+        run: |
+          echo webhook url $SLACK_WEBHOOK
+#         TODO: slack integration
```
Comment (trevor-scheer) [inline .github/workflows/publish.yml:18]: Is this still a TODO?

**Review by trevor-scheer — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 3 above)
Comment (clenfest) [reply .github/workflows/publish.yml:18]: I'll leave it for now. Either we keep the approval step and have to add slack integration or we get rid of it and have to make other changes.

**Pull request merged**